nav2_costmap_2d/include/nav2_costmap_2d/costmap_topic_collision_checker.hpp

```diff
@@ -69,8 +78,12 @@ class CostmapTopicCollisionChecker
 protected:
   /**
    * @brief Get a footprint at a set pose
+   *
+   * @param pose Pose to get footprint at
+   * @param fetch_latest_footprint Defaults to true. When checking with multiple poses at once,
+   * footprint should be fetched in the first check but fetching can be skipped in consequent checks for speedup
    */
-  Footprint getFootprint(const geometry_msgs::msg::Pose2D & pose, bool fetch);
+  Footprint getFootprint(const geometry_msgs::msg::Pose2D & pose, bool fetch_latest_footprint);
```

Comment:
```suggestion
  Footprint getFootprint(const geometry_msgs::msg::Pose2D & pose, bool fetch_latest_footprint = true);
```

I think you just need the default field and LGTM.